app/feature/feature-travel-certificate/src/main/kotlin/com/hedvig/android/feature/travelcertificate/ui/TravelCertificateHistory.kt

```diff
@@ -0,0 +1,422 @@
+package com.hedvig.android.feature.travelcertificate.ui
+
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.material.icons.Icons
+import androidx.compose.material3.CircularProgressIndicator
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Surface
+import androidx.compose.material3.Text
+import androidx.compose.material3.rememberModalBottomSheetState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import com.hedvig.android.core.designsystem.component.button.HedvigSecondaryContainedButton
+import com.hedvig.android.core.designsystem.component.information.HedvigInformationSection
+import com.hedvig.android.core.designsystem.preview.HedvigPreview
+import com.hedvig.android.core.designsystem.theme.HedvigTheme
+import com.hedvig.android.core.icons.Hedvig
+import com.hedvig.android.core.icons.hedvig.normal.Info
+import com.hedvig.android.core.ui.clearFocusOnTap
+import com.hedvig.android.core.ui.dialog.ErrorDialog
+import com.hedvig.android.core.ui.infocard.VectorInfoCard
+import com.hedvig.android.core.ui.scaffold.HedvigScaffold
+import com.hedvig.android.data.travelcertificate.TravelCertificate
+import com.hedvig.android.feature.travelcertificate.CertificateHistoryEvent
+import com.hedvig.android.feature.travelcertificate.CertificateHistoryUiState
+import com.hedvig.android.feature.travelcertificate.CertificateHistoryViewModel
+import com.hedvig.android.feature.travelcertificate.data.TravelCertificateUri
+import hedvig.resources.R
+import kotlinx.datetime.LocalDate
+
+@Composable
+internal fun TravelCertificateHistoryDestination(
+  viewModel: CertificateHistoryViewModel,
+  onContinue: () -> Unit,
```

Comment:
As discussed here https://hedviginsurance.slack.com/archives/C03U9C6Q7TP/p1707394659436749?thread_ts=1707390537.972059&cid=C03U9C6Q7TP I think we will soon have to hide this button all the way down where we call it if we do not want them to be able to create a new certificate right?

Probably something in the `uiState` itself actually can determine if this button should be shown or not.